### DISCUSSION TITLE
I have a NextJS Component. How do I make it a *dynamic* NextJS Component

### DISCUSSION THREAD

**Discussion by aumfer-uship** [vercel/next.js#48637] [Help]
### Summary

I have a NextJS component like this

```export default function MyComponent(){return<></>}```

When trying to add `getStaticPaths` I receive this error:

```Error: getStaticPaths can only be used with dynamic pages```

What do I do?

**Comment by haydenbleasel** [comment]
@aumfer-uship What is the filename and path of this Next.js page?

Essentially it's erroring because the path needs to look something like this:

```
.../[variable]/index.tsx
```
